packages/server/dataloader/atlassianLoaders.ts

```diff
@@ -2,16 +2,15 @@ import DataLoader from 'dataloader'
 import {decode} from 'jsonwebtoken'
 import JiraIssueId from 'parabol-client/shared/gqlIds/JiraIssueId'
 import {JiraGetIssueRes, JiraProject} from 'parabol-client/utils/AtlassianManager'
-import getAtlassianAuthByUserIdTeamId, {
-  AtlassianAuth
-} from '../postgres/queries/getAtlassianAuthByUserIdTeamId'
+import {AtlassianAuth} from '../postgres/queries/getAtlassianAuthByUserIdTeamId'
```

Comment:
This is a bit concerning. As we're adding more and more queries, we end up having a lot of redundant generated types. How do we decide which one to use as a database layer type? For example, I've added `getAtlassianAuthsByUserId` which is basically the same type as generated by `getAtlassianAuthByUserIdTeamId`. I'm afraid this could go out of control and it'd be nice to agree on how we handle that.

Reply:
i definitely share your worries as well. for this PR, i think it would be good to refactor to remove the single `upsertAtlassianAuth` to avoid repetition. to really address the concern, i think we should start a discussion in checkin meeting/slack or open a separate issue.  curious to hear matt's thoughts here.

Reply:
I would put the canonical version (i.e. get...ByUserId) into postgres/types and declare this as the return type. That's what I did for User anyway.

Reply:
as far as type systems go, typescript is structural (vs nominal) so any object that has the same shape as the canonical version will still fulfill the contract.

so, for example, if `byUserId` exports `AtlassianAuth` then the return value of `getAtlassianAuthByUserIdTeamId` will work in all the functions that accept `AtlassianAuth`.
if we wanted to be fancy, the return type of `getAtlassianAuthByUserIdTeamId` could `extend AtlassianAuth` which would be a way to point to the canonical version without a runtime penalty or code comment